pyproject.toml

```diff
@@ -7,11 +7,12 @@ readme = "README.md"
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.9"
+python = ">=3.9,<3.9.7 || >3.9.7,<4.0"
```

Comment:
why this version change?

Reply:
Source: https://github.com/streamlit/streamlit/blob/develop/lib/setup.py

# We exclude Python 3.9.7 from our compatible versions due to a bug in that version
# with typing.Protocol. See https://github.com/streamlit/streamlit/issues/5140 and
# https://bugs.python.org/issue45121